esp-hal/src/clock/mod.rs

```diff
@@ -341,56 +310,50 @@ pub struct RawClocks {
     pub pll_96m_clock: HertzU32,
 }
 
-cfg_if::cfg_if! {
-    if #[cfg(esp32c2)] {
-        static XTAL_FREQ_MHZ: AtomicU32 = AtomicU32::new(40);
+static mut ACTIVE_CLOCKS: Option<Clocks> = None;
 
-        pub(crate) fn xtal_freq_mhz() -> u32 {
-            XTAL_FREQ_MHZ.load(Ordering::Relaxed)
-        }
-    } else if #[cfg(esp32h2)] {
-        pub(crate) fn xtal_freq_mhz() -> u32 {
-            32
-        }
-    } else if #[cfg(any(esp32, esp32s2))] {
-        // Function would be unused
-    } else {
-        pub(crate) fn xtal_freq_mhz() -> u32 {
-            40
-        }
+impl Clocks {
+    pub(crate) fn init(cpu_clock_speed: CpuClock) {
+        critical_section::with(|_| {
+            unsafe { ACTIVE_CLOCKS = Some(Self::configure(cpu_clock_speed)) };
+        })
     }
-}
 
-/// Used to configure the frequencies of the clocks present in the chip.
-///
-/// After setting all frequencies, call the freeze function to apply the
-/// configuration.
-pub struct ClockControl {
-    desired_rates: RawClocks,
-}
+    fn try_get() -> Option<&'static Clocks> {
+        unsafe { ACTIVE_CLOCKS.as_ref() }
```

Comment:
We should probably write a safety comment here about how it's okay to use unsafe as it's not possible to change the clocks after calling `init`, meaning this will never be changed from underneath us

Reply:
Except it wouldn't be true, as embassy calls time_driver::now once before init 😭

Reply:
Which is, you know, same thread, same context, whatever, just...

Reply:
Does it? Doesn't embassy init occur after esp_hal's init?